### PR TITLE
Fix swagger specification mismatches

### PR DESCRIPTION
**Pull request by shiqizng**
## Solution

Adding swagger response annotations so a valid swagger doc generated.

## Testing

Used the existing `make generate` as a testing method.

### CODE REVIEW

**Comment by tsachiherman** [comment]
running `make generate` against master, I get:
```
2021/03/22 23:20:32
The swagger spec at "../swagger.json" is valid against swagger specification 2.0
2021/03/22 23:20:32
The swagger spec at "../swagger.json" showed up some valid but possibly unwanted constructs.
2021/03/22 23:20:32 See warnings below:
2021/03/22 23:20:32 - WARNING: response "#/responses/RawBlockResponse" is not used anywhere

The swagger spec at "./swagger.json" is invalid against swagger specification 2.0.
See errors below:
- some references could not be resolved in spec. First found: object has no key "WalletInfoResponse"

daemon/kmd/api/api.go:62: running "swagger": exit status 1
make: *** [generate] Error 1
```

while running against your PR, I get:
```
2021/03/22 23:21:08
The swagger spec at "../swagger.json" is valid against swagger specification 2.0
2021/03/22 23:21:08
The swagger spec at "../swagger.json" showed up some valid but possibly unwanted constructs.
2021/03/22 23:21:08 See warnings below:
2021/03/22 23:21:08 - WARNING: response "#/responses/RawBlockResponse" is not used anywhere
2021/03/22 23:21:12
The swagger spec at "./swagger.json" is valid against swagger specification 2.0
2021/03/22 23:21:12
The swagger spec at "./swagger.json" showed up some valid but possibly unwanted constructs.
2021/03/22 23:21:12 See warnings below:
2021/03/22 23:21:12 - WARNING: definition "#/definitions/APIV1RequestEnvelope" is not used anywhere
```

File: daemon/kmd/lib/kmdapi/responses.go [hunk 1]
```diff
@@ -54,95 +54,194 @@ type VersionsResponse struct {
...
+}
+
 // APIV1POSTWalletInfoResponse is the response to `POST /v1/wallet/info`
 // friendly:WalletInfoResponse
 type APIV1POSTWalletInfoResponse struct {
 	APIV1ResponseEnvelope
 	WalletHandle APIV1WalletHandle `json:"wallet_handle"`
 }
 
+// Response to `POST /v1/wallet/rename`
```
Comment (algorandskiy) [inline daemon/kmd/lib/kmdapi/responses.go:155]: rename -> info ?

**Review by algorandskiy — COMMENTED** [review]
Error is gone, good. Do we have a goal to fix warnings as well, or only error?

File: daemon/kmd/lib/kmdapi/responses.go (hunk 1 above)
Comment (shiqizng) [reply daemon/kmd/lib/kmdapi/responses.go]: the typo is fixed. I also put in a quick fix for the first warning.  I see that swagger:ignore annotation was already added for for APIV1RequestEnvelope but the definition still gets created in swagger.json. I'm unable to fix this warning at this time. swagger.json is still valid with this redundant definition.

**Comment by algorandskiy** [comment]
Looks like you can easily fix the last one about `APIV1RequestEnvelope` ?

**Comment by shiqizng** [comment]
> Looks like you can easily fix the last one about `APIV1RequestEnvelope` ?
@algorandskiy Unfortunately, I don't know a quick fix for this one. https://github.com/algorand/go-algorand/blob/master/daemon/kmd/lib/kmdapi/requests.go#L30-L33. As you can see there, swagger:ignore is already added but it's not behaving as expected.

**Comment by tsachiherman** [comment]
try to change the following:
```
// swagger:model VersionsRequest
type APIV1RequestEnvelope struct {
...
```

All swagger want to know is that you associate that with some command, and it's not being left abandoned.

**Comment by shiqizng** [comment]
thanks @tsachiherman. That worked. Both of the warnings are removed.

**Review by algorandskiy — APPROVED** [review]

**Review by tsachiherman — APPROVED** [review]

**Pull request merged**